File: src/expression/presence.py
"""Hold de presence para evitar flicker ao cruzar mãos."""


class PresenceHold:
    def __init__(self, hold_frames: int = 8) -> None:
        self._hold = hold_frames
        self._missing = 0
        self._seen = False

    def update(self, present: bool) -> bool:
        if present:
            self._missing = 0
            self._seen = True
            return True
        self._missing += 1
        if self._seen and self._missing < self._hold:
            return True
        self._seen = False
        return False

File: packages/mapping/src/presence.ts
export class PresenceHold {
  private missing = 0;
  private seen = false;

  constructor(private hold = 8) {}

  update(present: boolean): boolean {
    if (present) {
      this.missing = 0;
      this.seen = true;
      return true;
    }
    this.missing += 1;
    if (this.seen && this.missing < this.hold) return true;
    this.seen = false;
    return false;
  }
}
